packages/tables/src/Concerns/CanSortRecords.php

```diff
@@ -97,24 +98,24 @@ protected function applySortingToTableQuery(Builder $query): Builder
 
     protected function applyDefaultSortingToTableQuery(Builder $query): Builder
     {
-        $sortColumnName = $this->getTable()->getDefaultSortColumn();
+        $defaultSort = $this->getTable()->getDefaultSort();
         $sortDirection = ($this->getTable()->getDefaultSortDirection() ?? $this->tableSortDirection) === 'desc' ? 'desc' : 'asc';
 
         if (
-            $sortColumnName &&
-            ($sortColumn = $this->getTable()->getSortableVisibleColumn($sortColumnName))
+            is_string($defaultSort) &&
+            ($sortColumn = $this->getTable()->getSortableVisibleColumn($defaultSort))
         ) {
             $sortColumn->applySort($query, $sortDirection);
 
             return $query;
         }
 
-        if ($sortColumnName) {
-            return $query->orderBy($sortColumnName, $sortDirection);
+        if (is_string($defaultSort)) {
+            return $query->orderBy($defaultSort, $sortDirection);
         }
 
-        if ($sortQueryUsing = $this->getTable()->getDefaultSortQuery()) {
-            app()->call($sortQueryUsing, [
+        if ($defaultSort instanceof Closure) {
+            app()->call($defaultSort, [
                 'direction' => $sortDirection,
                 'query' => $query,
             ]);
```

Comment:
At the moment, the default sort function is executed twice, first it is evaluated and then it is run again here. Can we instead pass the direction and query into the `getDefaultSort()` function, so we don't have to run it again?

Reply:
I think I implemented this the way you are thinking, but let me know if you meant something else